notebook/agentchat_hierarchy_flow_using_select_speaker.ipynb

```diff
@@ -0,0 +1,397 @@
+    "                except ValueError:\n",
+    "                    pass  # If agent name is not valid, continue with normal selection\n",
+    "            elif 'TERMINATE' in last_message['content']:\n",
+    "                try:\n",
+    "                    return self.agent_by_name('User_proxy')\n",
+    "                except ValueError:\n",
+    "                    pass  # If 'User_proxy' is not a valid name, continue with normal selection\n",
+    "        \n",
+    "        team_leader_names = [agent.name for agent in self.agents if agent.name.endswith('1')]\n",
+    "\n",
+    "        if last_speaker.name in team_leader_names:\n",
+    "            team_letter = last_speaker.name[0]\n",
+    "            possible_next_speakers = [\n",
+    "                agent for agent in self.agents if (agent.name.startswith(team_letter) or agent.name in team_leader_names) \n",
+    "                and agent != last_speaker and agent != self.previous_speaker\n",
+    "            ]\n",
+    "        else:\n",
+    "            team_letter = last_speaker.name[0]\n",
+    "            possible_next_speakers = [\n",
+    "                agent for agent in self.agents if agent.name.startswith(team_letter) \n",
+    "                and agent != last_speaker and agent != self.previous_speaker\n",
+    "            ]\n",
+    "\n",
+    "        self.previous_speaker = last_speaker\n",
+    "\n",
+    "        if possible_next_speakers:\n",
+    "            next_speaker = random.choice(possible_next_speakers)\n",
+    "            return next_speaker\n",
+    "        else:\n",
+    "            return None\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Termination message detection\n",
+    "def is_termination_msg(content) -> bool:\n",
+    "    have_content = content.get(\"content\", None) is not None\n",
+    "    if have_content and \"TERMINATE\" in content[\"content\"]:\n",
+    "        return True\n",
+    "    return False"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[33mB2\u001b[0m (to chat_manager):\n",
+      "\n",
+      "Find the product of x and y, the other agents know x and y.\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n",
+      "[autogen.oai.completion: 10-31 09:38:16] {788} WARNING - Completion was provided with a config_list, but the list was empty. Adopting default OpenAI behavior, which reads from the 'model' parameter instead.\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:autogen.oai.completion:Completion was provided with a config_list, but the list was empty. Adopting default OpenAI behavior, which reads from the 'model' parameter instead.\n"
```

Comment:
This warning implies that the config_list is not loaded properly. Is it intentional?

Reply:
I have corrected the path and there is no more warning.
<img width="945" alt="image" src="https://github.com/microsoft/autogen/assets/14764966/770b7fe4-537d-48c9-8804-73af61654a80">